manifests/init.pp

```diff
@@ -71,6 +71,24 @@
         notify  => Service[$servicename],
         content => template('dhcp/redhat/sysconfig-dhcpd'),
       }
+      if $::operatingsystemrelease =~ /^7/ {
+        file{ '/etc/systemd/system/dhcpd.service':
+          ensure  => file,
+          owner   => 'root',
+          group   => 'root',
+          mode    => '0644',
+          require => Package[$packagename],
+          notify  => Exec['daemon-reload'],
+          content => template('dhcp/redhat/dhcpd.service'),
+        }
+        exec { 'daemon-reload':
```

Comment:
Include the module name here to prevent conflicts, e.g. `systemctl-daemon-reload-dhcp` to match the puppet module.